STD-31 | Tag commands\queries

zefirlover opened the pull request.

zefirlover committed:
RemoveTagCommand.cs was added

WerewolfNet commented on server/StudySharp.ApplicationServices/Commands/RemoveTagCommand.cs:
```diff
@@ -0,0 +1,33 @@
+    {
+        private readonly StudySharpDbContext _studySharpDbContext;
+
+        public async Task<OperationResult> Handle(RemoveTagCommand request, CancellationToken cancellationToken)
```
1) RemoveTag => RemoveTagById
2) Нужно вытягивать Tag по id, проверять, null ли он, и если он не null, то производить удаление

zefirlover replied:
Ти хоч шоб імя класу я замінив із RemoveTagCommand на RemoveTagByIdCommand чи на RemoveTagById?

WerewolfNet replied:
RemoveTagByIdCommand

WerewolfNet commented on server/StudySharp.ApplicationServices/Commands/RemoveTagCommand.cs:
```diff
@@ -0,0 +1,33 @@
+{
+    public sealed class RemoveTagCommand : IRequest<OperationResult>
+    {
+        public string Name { get; set; }
```
Удаляем пока что по id.
Name => id

WerewolfNet commented on server/StudySharp.ApplicationServices/Commands/RemoveTagCommand.cs:
```diff
@@ -0,0 +1,33 @@
+
+        public async Task<OperationResult> Handle(RemoveTagCommand request, CancellationToken cancellationToken)
+        {
+            if (await _studySharpDbContext.Tags.AnyAsync(_ => _.Name.ToLower() != request.Name.ToLower()))
```
в .AnyAsync вторым параметром прокинь cancellationToken

zefirlover replied:
Другий параметр це той шо замість request.Name.ToLower()?

WerewolfNet replied:
Посмотри ПР, который связан был с Course

WerewolfNet commented on server/StudySharp.ApplicationServices/Commands/RemoveTagCommand.cs:
```diff
@@ -0,0 +1,33 @@
+            }
+
+            _studySharpDbContext.Tags.Remove(new Tag { Name = request.Name });
+            await _studySharpDbContext.SaveChangesAsync();
```
в .SaveChangesAsync прокинь cancellationToken

WerewolfNet replied:
Тоже самое (попрокидовать везде cancellationToken в Async методах) сделай и с AddTagCommand, GetTagByIdQuery

WerewolfNet commented on server/StudySharp.ApplicationServices/Commands/RemoveTagCommand.cs:
```diff
@@ -0,0 +1,33 @@
+                return OperationResult.Fail(string.Format(ErrorConstants.EntityNotFound, nameof(Tag), nameof(Tag.Name), request.Name));
+            }
+
+            _studySharpDbContext.Tags.Remove(new Tag { Name = request.Name });
```
Это что такое?)

zefirlover committed:
Major changes in RemoveTagCommand.cs
RemoveTagCommand.cs was renamed to RemoveTagByIdCommand.cs
cancellationToken was added to tag in GetTagByIdQuery.cs, to if statement condition and SaveChangesAsync in AddTagCommand.cs and to tag + SaveChangesAsync in RemoveTagByIdCommand.cs
RemoveTagByIdCommand.cs was partly reworked

zefirlover committed:
Unnecessary using was removed in RemoveTagByIdCommand.cs
